### PR TITLE
Clarification and Consensus on Column Requirements

### PR DESCRIPTION
**Pull request by jpradocueva**
This PR must be reviewed alongside the `columns`  tab in the spreadsheet titled: [FOCUS Columns - Required vs Not Required](https://docs.google.com/spreadsheets/d/1eWILtXdZo9Y1aNOXqmAV0PTGm5IGF9E9oXvJckP81dw/edit?usp=sharing).

This PR introduces changes to align stakeholder consensus on column requirements in the project's documentation. It addresses concerns around terms like compliance and essentials, aiming to clarify MANDATORY and OPTIONAL columns. These adjustments aim to enhance clarity across teams. 

If these changes are acceptable, I will create a PR with a table listing which columns are mandatory and optional.

### CODE REVIEW

**Comment by jpradocueva** [comment]
If we want to create a roadmap or journey for companies that may not yet meet all mandatory features may be a good strategy for FOCUS release 1.0 It acknowledges that not all organizations can achieve full compliance immediately but encourages continuous improvement towards achieving all critical standards. 

By structuring the implementation of features into the categories listed below, companies can have a clearer vision of the path to full compliance or feature completeness. It helps in setting realistic goals, managing stakeholder expectations, and prioritizing development efforts. Additionally, this approach enables companies to demonstrate commitment to continuous improvement and compliance, which can be crucial for regulatory approval, market acceptance, and competitive advantage.

These are some alternatives to categorize features into a structured roadmap, facilitating a gradual approach to full compliance or feature implementation:

Implementation Phases:
Phase 1: Immediate Implementation - Mandatory features that need to be implemented as a priority for the project to function or to meet minimal legal requirements.
Phase 2: Short-term Goals - Important features that are not immediately critical but should be implemented in the near term to improve efficiency, security, or user satisfaction.
Phase 3: Long-term Enhancements - Optional features or enhancements that can be considered over time to add value or stay ahead of the competition.

Priority Tiers:
Tier 1: Must-Have - Critical features required for minimum viable product (MVP) or compliance.
Tier 2: Should-Have - Features that are not critical but are important for a fuller version of the product.
Tier 3: Could-Have - Features that would be nice to include if additional resources or time become available.

Development Sprints:
Sprint 0: Foundation - Setting up the basic infrastructure and implementing the most critical mandatory features.
Sprint 1: Core Functionality - Building out essential features that form the core of the product or compliance requirements.
Sprint 2+: Feature Expansion - Gradually adding optional features in order of priority, based on user feedback, market needs, or regulatory changes.

Maturity Model:
Initiation: Recognition of essential features and initial efforts to implement mandatory requirements.
Development: Active development and implementation of both mandatory and selected optional features.
Optimization: Refinement and optimization of implemented features, plus the addition of new features to enhance capabilities and compliance.

File: specification/columns/availabilityzone.md
Comment (flanakin) [inline specification/columns/availabilityzone.md:5]: I believe this should be RECOMMENDED. The intent was to not require AvailabilityZone for the 1.0 timeframe. 
```suggestion
The AvailabilityZone column is RECOMMENDED to be present in the billing data when the provider supports deploying resources or services within an *availability zone*. This column MUST be of type String and MAY contain null values when a charge is not specific to an *availability zone*.
```

File: specification/columns/region.md
Comment (flanakin) [inline specification/columns/region.md:1]: Shouldn't this file be `regionname.md`?

File: specification/columns/resourceid.md
Comment (flanakin) [inline specification/columns/resourceid.md:5]:
```suggestion
The ResourceId column is OPTIONAL, but MUST be present in the billing data when the provider supports billing based on provisioned resources. This column MUST be of type String. The ResourceId value MAY be a nullable column as some cost data [*rows*](#glossary:row) may not be associated with a *resource*. ResourceId MUST appear in the cost data if an identifier is assigned to a *resource* by the provider. ResourceId SHOULD be a fully-qualified identifier that ensures global uniqueness within the provider.
```

File: specification/columns/availabilityzone.md
Comment (thecloudman) [reply specification/columns/availabilityzone.md:5]: My view is that we keep the MUST, SHOULD wording. Allow NULL and add some information around, if you dont have this column or data related to this column, either add the column with NULL or exclude the column.

File: specification/columns/availabilityzone.md
Comment (thecloudman) [inline specification/columns/availabilityzone.md:5]: The "AvailabilityZone" column MUST be included in the billing data whenever the provider supports resources within an availability zone. This column:

•	MUST always be of type string. 
•	MAY contain NULL values if a charge is not associated with a specific availability zone.

This ensures data consistency and accuracy when tracking billing information related to resource usage in various availability zones.

**Review by thecloudman — CHANGES_REQUESTED** [review]

File: specification/columns/availabilityzone.md
Comment (flanakin) [reply specification/columns/availabilityzone.md:5]: Alternatively we could use MAY instead of MUST here. That's duplicative of OPTIONAL, but conveys both points. RECOMMENDED or SHOULD is probably simpler.
```suggestion
The AvailabilityZone column is OPTIONAL, but MAY be present in the billing data when the provider supports deploying resources or services within an *availability zone*. This column MUST be of type String and MAY contain null values when a charge is not specific to an *availability zone*.
```

Comment (projectlocale) [reply specification/columns/availabilityzone.md:5]: Mostly agree with this, except the implementation of the idea that I think has the most clarity from an adoption perspective and the least friction would be to use create a default value such that if the value is not flipped certain assumptions are made about the conditions which can be changed by specifying just what the user wants to change.

**Review by marc-perreaut — APPROVED** [review]
I am in line with the usage of MUST, SHOULD, and MAY, especially the usage of SHOULD for the columns that not all providers are in position to support. That makes FOCUS more inclusive.

About the Column ID, I am fine with "Feature Level = Mandatory/Conditional/Optional".

Another option to be closer to https://datatracker.ietf.org/doc/html/rfc2119 would be "Requirement Level = Mandatory / Recommended / Optional". But that's minor.

**Comment by jpradocueva** [comment]
@marc-perreaut, you are right; we could use `Recommended` instead of `Conditional.` I intend to highlight that those features marked as `Conditional` need further discussion without any presumption of what the outcome could be. If the preference is `Recommended,` I am also okay with that as we will have to revisit this topic once the PR is merged.

File: specification/columns/chargefrequency.md
Comment (jpradocueva) [inline specification/columns/chargefrequency.md:5]:
```suggestion
The ChargeFrequency column MUST be present in the billing data and MUST NOT be null. This column is of type String and MUST be one of the allowed values. When [ChargeCategory](#chargecategory) is "Purchase", ChargeFrequency MUST NOT be "Usage-Based".
```

File: specification/columns/chargefrequency.md
Comment (jpradocueva) [reply specification/columns/chargefrequency.md:25]: This is link to #371. It requires further discussion.

File: specification/columns/chargefrequency.md
Comment (jpradocueva) [inline specification/columns/chargefrequency.md:24]:
```suggestion
| Feature Level   | Mandatory       |
```

File: specification/columns/chargesubcategory.md
Comment (jpradocueva) [reply specification/columns/chargesubcategory.md:13]: The group agreed to review all the SHOULD and Feature Levels set to Conditional once PR #331 is merged.

**Comment by jpradocueva** [comment]
The Maintainers agreed to merge this PR into PR #331. The group agreed to review any column defined as "Conditional," where the requirements are set to `SHOULD.` Also, we agreed to keep `ChargeFrequency` and `Availability` as they were. 
ChargeFrequency has pending Issue #371 to make the column required under certain conditions.

**Pull request merged**
